src/MPQEditor/MPQCircleSelector.ts

```diff
@@ -201,13 +230,31 @@ export class MPQSelectionPanel
     this._modelPath = args.modelPath;
     this._mpqEventHandler = handler;
     this._lastSelected = args.names;
+    this._visqData = [];
 
     // Listen for when the panel is disposed
     // This happens when the user closes the panel or when the panel is closed programmatically
     this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
 
     this.initGraphCtrl(this._modelPath, this);
-    this.setMode("selector");
+    if (visqPath.length < 1) {
+      this.setMode("selector");
+    } else {
+      this.setMode("visqselector");
+      const visqUri = vscode.Uri.file(visqPath);
+      vscode.workspace.fs.readFile(visqUri).then((visqData) => {
+        try {
+          this._visqData = JSON.parse(visqData.toString());
+        } catch (error) {
+          this.onInvalidVISQData(visqPath);
+        }
+
+        // check whether _visqData pretend to be valid
+        if (!("error" in this._visqData) || !("meta" in this._visqData)) {
+          this.onInvalidVISQData(visqPath);
```

Comment:
Q. Isn't it expected to be valid here?

Reply:
The condition says, that `_visqData` should contain `error` section and `meta` section, otherwise (doesn't contain `error` or doesn't contain `meta`) it is treated as invalid.

Reply:
The condition was meant to prevent user from loading other file, so it is very simple.

Reply:
@dayo09
The simplest form for _visqData to be valid is that it contains both "error" and "meta", otherwise it can be treated as invalid.
So negation of 
```
("error" in _visqData) and ("meta" in _visqData)
```
gives us
```
("eror" isn't in _visqData) or ("meta" isn't in _visqData)
```
i.e. if  _visqData doesn't contain either "error" or "meta" it can be treated as invalid.

Reply:
@dayo09 
Do i need to remove this condition for now?

Reply:
Aha... I didn't know that. What I thought was: if it includes 'error', than isn't it 'invalid' itself?

Reply:
> Aha... I didn't know that. What I thought was: if it includes 'error', than isn't it 'invalid' itself?

No. It's valid. Valid `visq.json` file should contain `error` along with `meta` section at least.

Reply:
Ah, sorry that now I understand it. So 'Error' is the error ratio. I thought it's a problematic log. Thanks for the explanation.